src/java.base/share/classes/java/lang/snippet-files/ProcessExamples.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.Reader;
+import java.io.Writer;
+
+public class ProcessExamples {
+    // @start region=example
+    void example() {
```

Comment:
Should there be another version of this that is involved with interruption set at top of method? That example would forever freeze which shows there is a problem with the example.

Reply:
I don't see a hang in this example, it writes 3 lines to the child and flushes and closes the output stream.
`cat` will see end of file and exit.
Almost all processes started will exit when they've done what they are supposed to.
A more defensive example would give the impression that every use of a process should follow the same pattern.
It adds a lot of boilerplate to schedule an interrupt and then cancel when the process exits.